nixos/modules/services/networking/v2ray.nix

```diff
@@ -83,13 +83,11 @@ with lib;
       }
     ];
 
-    systemd.services.v2ray = {
-      description = "v2ray Daemon";
-      after = [ "network.target" ];
-      wantedBy = [ "multi-user.target" ];
-      serviceConfig = {
-        ExecStart = "${cfg.package}/bin/v2ray -config ${configFile}";
-      };
-    };
+    environment.etc."v2ray/config.json".source = configFile;
+
+    systemd.packages = [ cfg.package ];
+
+    # Workaround: https://github.com/NixOS/nixpkgs/issues/81138
+    systemd.services.v2ray.wantedBy = [ "multi-user.target" ];
```

Comment:
Would the service still be restarted/reloaded on configuration changes with this approach? Maybe a reloadTrigger has to be added.

Reply:
Added `restartTriggers`. It doesn't seem to support online reloading.